### PR TITLE
test(core): move rewards tests from legacy-integration to bats

### PR DESCRIPTION
**Pull request by vindard**

### CODE REVIEW

File: bats/core/api/earn.bats [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  cache_funder_wallet_id
+  fund_wallet_intraledger \
+    'alice' \
+    "alice.btc_wallet_id" \
+    "funder.btc_wallet_id" \
+    "10000"
+
+}
+
+@test "onchain-receive: btc wallet, can create new address if current one is unused" {
```
Comment (dolcalmi) [inline bats/core/api/earn.bats:20]: is this the intended name?

File: bats/core/api/earn.bats [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  [[ "${error_msg}" =~ " already claimed" ]] || exit 1
+
+  # Check balance after retry
+  exec_graphql $token_name 'wallets-for-account'
+  btc_balance_after_retry=$(graphql_output '
+    .data.me.defaultAccount.wallets[]
+    | select(.walletCurrency == "BTC")
+    .balance
+  ')
+  [[ "$btc_balance_after_retry" == "$btc_balance_after_earn" ]] || exit 1
```
Comment (dolcalmi) [inline bats/core/api/earn.bats:67]: we are missing memo validation

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: bats/core/api/earn.bats (hunk 1 above)
Comment (vindard) [reply bats/core/api/earn.bats:20]: Bad copy-paste, thanks!

File: bats/core/api/earn.bats (hunk 2 above)
Comment (vindard) [reply bats/core/api/earn.bats:72]: Added in [`force-push`](https://github.com/GaloyMoney/galoy/compare/3bc189b9189d4f29c84930b135f281a7359175c1..b2b7d32759832f4d208ef5fcbc2b5ea874002b53)

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**